runtime_scan/pkg/orchestrator/scanwatcher/watcher.go

```diff
@@ -0,0 +1,220 @@
+	pollPeriod       time.Duration
+	reconcileTimeout time.Duration
+}
+
+func (w *Watcher) Start(ctx context.Context) {
+	queue := common.NewQueue[ScanReconcileEvent]()
+
+	poller := &ScanPoller{
+		Logger:     w.logger,
+		PollPeriod: w.pollPeriod,
+		Queue:      queue,
+		GetItems:   w.GetAbortedScans,
+	}
+	poller.Start(ctx)
+
+	reconciler := &ScanReconciler{
+		ReconcileTimeout:  w.reconcileTimeout,
+		Queue:             queue,
+		ReconcileFunction: w.Reconcile,
+	}
+	reconciler.Start(ctx)
+}
+
+func (w *Watcher) GetAbortedScans(ctx context.Context) ([]ScanReconcileEvent, error) {
+	scans, err := w.getScansByState(ctx, models.ScanStateAborted)
+	if err != nil || scans.Items == nil || len(*scans.Items) <= 0 {
+		return nil, err
+	}
+
+	count := len(*scans.Items)
+	r := make([]ScanReconcileEvent, count)
+	for i, scan := range *scans.Items {
+		r[i] = ScanReconcileEvent{
+			ScanID: *scan.Id,
+		}
+	}
+
+	return r, nil
+}
+
+func (w *Watcher) Reconcile(ctx context.Context, event ScanReconcileEvent) error {
+	w.logger.Infof("reconciling scan event: %v", event)
+
+	selector := "id,state,stateReason"
+	params := models.GetScansScanIDParams{
+		Select: &selector,
+	}
+
+	scan, err := w.client.GetScan(ctx, event.ScanID, params)
+	if err != nil || scan == nil {
+		return fmt.Errorf("getting scan with id %s failed: %v", event.ScanID, err)
+	}
+
+	state, ok := scan.GetState()
+	if !ok {
+		return fmt.Errorf("cannot determine state of Scan with %s id", event.ScanID)
+	}
+
+	switch state {
+	case models.ScanStateDone, models.ScanStateFailed:
+		w.logger.Infof("reconciling scan event is skipped as Scan is already finished: %v", event)
+	case models.ScanStateAborted:
+		return w.reconcileAborted(ctx, event)
+	case models.ScanStatePending, models.ScanStateDiscovered, models.ScanStateInProgress:
+		fallthrough
+	default:
+	}
+
+	return nil
+}
+
+func (w *Watcher) getScansByState(ctx context.Context, s models.ScanState) (models.Scans, error) {
+	filter := fmt.Sprintf("state eq '%s'", s)
+	selector := "id"
+	params := models.GetScansParams{
+		Filter: &filter,
+		Select: &selector,
+	}
+	scans, err := w.client.GetScans(ctx, params)
+	if err != nil {
+		err = fmt.Errorf("getting Scan(s) by their state failed: %v", err)
+	}
+	if scans == nil {
+		scans = &models.Scans{}
+	}
+
+	return *scans, err
+}
+
+func (w *Watcher) reconcileAborted(ctx context.Context, event ScanReconcileEvent) error {
+	filter := fmt.Sprintf("scan/id eq '%s' and status/general/state ne '%s' and status/general/state ne '%s'",
+		event.ScanID, models.ABORTED, models.DONE)
+	selector := "id,status"
+	params := models.GetScanResultsParams{
+		Filter: &filter,
+		Select: &selector,
+	}
+
+	scanResults, err := w.client.GetScanResults(ctx, params)
+	if err != nil {
+		return fmt.Errorf("getting ScanResult(s) for Scan with %s id failed: %v", event.ScanID, err)
+	}
+
+	if scanResults.Items != nil && len(*scanResults.Items) > 0 {
+		var retryIsNeeded bool
```

Comment:
Not sure how retry is reflected here or to the user.

Reply:
@akpsgit The `retryIsNeeded` is used here to inform the the `common.Reconciler` that the reconciliation is failed.
So if your suggestion is that this needs to be a silent error (e.g. without logging/returning error) and let the `common.Poller` pick it up the aborted scan again in case the `retryIsNeeded == true` than it is fine by me.

Reply:
just when  (retryIsNeeded == true) nothing that is related to retry happening (e.g. to do a retry, notify in the error text that a retry is needed (to be done by the user?), etc.). 
or if we don't want to handle the retry now, maybe we can change the retryIsNeeded var to patchErr and concatenate it to the returned error. WDYT?

Reply:
So the confusion around `retryIsNeeded` was that the name of the variable which suggested that there is a retry logic which is triggered based on the value of this variable. It has been renamed to `reconciliationFailed` which provides a better explanation for the use of this variable.